src/kagglehub/auth.py

```diff
@@ -9,20 +12,96 @@
+NOTEBOOK_LOGIN_TOKEN_HTML_END = """
+<b>Thank You</b></center>"""
 
-    username = input("Enter your Kaggle username: ")
-    api_key = input("Enter your Kaggle API key: ")
 
-    set_kaggle_credentials(username=username, api_key=api_key)
+@contextmanager
+def capture_logger_output():
+    """Capture output that is logged using the logger.
+
+    Example:
+    ```py
+    >>> output = capture_logger_output()
+    >>> logger.info("This is an info message")
+    >>> logger.error("This is an error message")
+    >>> print(output)
+    ```
+    """
+    buffer = io.StringIO()
+    handler = logging.StreamHandler(buffer)
+    logger = logging.getLogger()
+    logger.addHandler(handler)
+    try:
+        yield buffer
+    finally:
+        logger.removeHandler(handler)
 
-    logger.info("Kaggle credentials set.")
 
-    if not validate_credentials:
-        return
+def is_in_notebook():
+    return "IPython" in sys.modules
+
+
+def notebook_login(validate_credentials) -> None:
+    """Prompt the user for their Kaggle token and save it in a widget (Jupyter or Colab)."""
+    library_error = "You need the `ipywidgets` module: `pip install ipywidgets`."
+    try:
+        from IPython.display import display  # type: ignore
+        from ipywidgets import widgets  # type: ignore
+    except ImportError:
+        raise ImportError(library_error)  # noqa: B904
+
+    box_layout = widgets.Layout(display="flex", flex_flow="column", align_items="center", width="50%")
+
+    username_widget = widgets.Text(description="Username:")
+    token_widget = widgets.Password(description="Token:")
+    login_button = widgets.Button(description="Login")
+
+    login_token_widget = widgets.VBox(
+        [
+            widgets.HTML(NOTEBOOK_LOGIN_TOKEN_HTML_START),
+            username_widget,
+            token_widget,
+            login_button,
+            widgets.HTML(NOTEBOOK_LOGIN_TOKEN_HTML_END),
+        ],
+        layout=box_layout,
+    )
+    display(login_token_widget)
+
+    # On click events
+    def login_token_event(t):  # noqa: ARG001
+        username = username_widget.value
+        token = token_widget.value
+        # Erase token and clear value to make sure it's not saved in the notebook.
+        token_widget.value = ""
+
+        # Hide inputs
+        login_token_widget.children = [widgets.Label("Connecting...")]
+        try:
+            with capture_logger_output() as captured:
+                # Set Kaggle credentials
+                set_kaggle_credentials(username=username, api_key=token)
 
+                # Validate credentials if necessary
+                if validate_credentials is True:
+                    validate_credentials_helper()
+            message = captured.getvalue()
+        except Exception as error:
+            message = str(error)
+        # Print result (success message or error)
+        login_token_widget.children = [widgets.Label(line) for line in message.split("\n") if line.strip()]
```

Comment:
Those this allow you to try again with a new set of credentials?

Reply:
no, you would have to run kagglehub.login() again in your notebook. if we make it automatically allow the user to login again, we would need to put a maximum number of retries i think? because else, if our server is down or the user forgot his credentials, it will keep popping up